source/includes/steps-tutorial-dotnet.yaml

```
@@ -0,0 +1,316 @@
+
+  The ``ProjectPage.xaml.cs`` file is the default screen for this app. In the 
+  ``OnStart()`` method, we check to see if we have an authenticated user 
+  already. If not, we redirect to the Login page. If we do, or when control is 
+  returned from the Login page, we can then instantiate a
+  :dotnet-sdk:`SyncConfiguration <reference/Realms.Sync.SyncConfiguration.html>` 
+  object and pass that to the 
+  :dotnet-sdk:`GetInstanceAsync() <reference/Realms.Realm.html#Realms_Realm_GetInstanceAsync_Realms_RealmConfigurationBase_System_Threading_CancellationToken_>` 
+  method to get the user's custom realm. From this realm, we fetch the specific 
+  user document that has an ``_id`` value matching the ``Id`` property. 
+  Each ``User`` object contains a read-only ``IList<Project>`` 
+  property called ``MemberOf``, and we iterate through this list to display the 
+  Projects in the UI. 
+
+  To implement this functionality, find the ``OnStart()`` method in 
+  ``ProjectPage.xaml.cs``. You see the conditional checking to see if we already 
+  have an authenticated user. In the ``else`` block (we have an authenticated user),
+  instantiate the ``userRealm`` object by creating a ``SyncConfiguration`` object 
+  and passing that to the ``Realm.GetInstanceAsync()`` method. When building the 
+  SyncConfiguration object, we set the partition string to ``user=<USER_ID>``:
+  
+  .. literalinclude:: /tutorial/generated/code/final/ProjectPage.codeblock.user-realm-config.cs
+     :language: csharp
+
+  We then find the user from the User realm that matches the authenticated user's
+  Id:
+
+  .. literalinclude:: /tutorial/generated/code/final/ProjectPage.codeblock.find-user.cs
+     :language: csharp
+
+  Once you have the user, call ``SetUpProjectList()``. In that method, you populate 
+  the ``MyProjects`` collection (an ``ObservableCollection<Project>``), which is 
+  used to keep the list of projects up-to-date.
+  .. implement SetUpProjectList
+  
+  When the app user touches one of the Projects in the list, we pass 
+  control to the ``TaskPage.xaml.cs`` class and display the list of Tasks within 
+  the selected Project. 
+
+---
+title: Implement the Task Page
+ref: implement-task-page
+content: |
+  Open the ``TaskPage.xaml.cs`` file. In the ``OnStart()`` method, we will create 
+  another realm, this time for the Task collection. In the user realm we used a 
+  partition key value of ``user=<USER_ID>``; the project realm uses a partition 
+  key value of ``project=<USER_ID>``. For the sake of simplicity in this tutorial, 
+  each user has exactly one project they own, and it is specified by the partition 
+  key value containing their own user ID:
+
+  .. literalinclude:: /tutorial/generated/code/final/TaskPage.codeblock.task-realm-config.cs
+     :language: csharp
+
+  In the ``SetUpTaskList()`` method, find all of the tasks. We find all tasks because 
+  we configured our taskRealm to use the partition specific to the user, removing 
```

```suggestion
  we configured our ``taskRealm`` to use the partition specific to the user, removing 
```

should taskRealm be monospaced since it is a variable?